### PR TITLE
Cut v2.42.0-rc.0

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # Changelog
 
+## 2.42.0-rc.0 / 2023-01-26
```
Comment (bwplotka) [inline CHANGELOG.md:3]: Hm, I am missing some changes worth mentioning. Could you double check?

e.g.

* Native histogram improvements:
   * https://github.com/prometheus/prometheus/pull/11864
  * Promtool: https://github.com/prometheus/prometheus/pull/11872

There might be more.

Comment (kakkoyun) [reply CHANGELOG.md:3]: I'll re-check the history

**Review by Nexucis — APPROVED** [review]
🥳

**Review by roidelapluie — APPROVED** [review]

**Comment by kakkoyun** [comment]
Merging it to tag and start benchmarking.

**Review by codesome — COMMENTED** [review]
@kakkoyun I did not notice this until the rc announcement. I would like to make the following changes to the CHANGELOG for the final release w.r.t. histograms, some of them I understand are a little hard to parse from the commits.

---

Some version of this note at the top to warn native histogram testers

```
This release comes with a bunch of feature coverage for native histograms and breaking changes.

If you are trying native histograms already, you are recommended to remove the `wal` directory when upgrading because the old WAL record for native histograms are not backward compatible in v2.42.0. This will lead to some data loss for the latest data.

Additionally, if you scrape "float histograms" or use recording rules on native histograms in v2.42.0 (which writes float histograms), it is a one-way street since older versions do not support float histograms.
```

---

A couple of new entries

```
* [CHANGE] **breaking** TSDB: Changed WAL record format for the experimental native histograms. #11783
* [ENHANCEMENT] Rules: Support native histograms in recording rules. #11838
```

---

Edit in these two entries

```

* [ENHANCEMENT] TSDB: Support FloatHistogram. #11522 #11817
=>
* [ENHANCEMENT] TSDB/Scrape: Support FloatHistogram that represents buckets as float64 values. #11522 #11817 #11716

* [ENHANCEMENT] TSDB: Add gauge histogram support. #11783 #11840
=>
* [ENHANCEMENT] TSDB: Add gauge histogram support. #11783 #11840 #11814
```

---

This is not user-facing at the moment, so should be excluded from the release notes:

```
* [ENHANCEMENT] Histograms: Return actually useful counter reset hints. #11864
```

**Comment by kakkoyun** [comment]
@codesome Thanks for noticing these. PTAL https://github.com/prometheus/prometheus/pull/11909
